openfisca_core/entities.py

```diff
@@ -98,16 +98,19 @@ def init_variable_values(self, entity_object, entity_id):
                         except KeyError:
                             possible_values = [item.name for item in holder.variable.possible_values]
                             raise SituationParsingError(path_in_json,
-                                "'{}' is not a valid value for '{}'. Possible values are ['{}'].".format(
+                                "'{}' is not a known value for '{}'. Possible values are ['{}'].".format(
                                     value, variable_name, "', '".join(possible_values))
                                 )
                     try:
                         array[entity_index] = value
+                    except (OverflowError):
+                        error_message = "Can't deal with value: '{}', it's too large for type {}.".format(value, holder.variable.value_type)
```

Comment:
This gives `{"2017":"Can't deal with value: '10000000000000000000', it's too large for type <class 'int'>."}` which doesn't read super well.